src/Psy/Plugin/Manager.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Psy\Plugin;
+
+class Manager
+{
+    /** @var AbstractPlugin[] */
+    protected static $plugins = array();
+
+    /**
+     * @param AbstractPlugin $plugin
+     * @param $name
+     */
+    public static function register(AbstractPlugin $plugin, $name)
+    {
+        self::$plugins[$name] = $plugin;
+    }
+
+    /**
+     * @param array $configuration
+     *
+     * @return array
+     */
+    public static function getConfiguration($configuration = array())
+    {
+        foreach (self::$plugins as $plugin) {
+            $configuration = $plugin::getConfiguration($configuration);
```

Comment:
oh, them I don't think it should merge anything, it should just return the brute configuration rather than try to merge it with something else

Even though it is really subjective, I think it could be confusing (at least I was confused :x)